components/use-paid-mutation.js

```diff
@@ -23,33 +25,86 @@ export function usePaidMutation (mutation,
   const [getPaidAction] = useLazyQuery(GET_PAID_ACTION, {
     fetchPolicy: 'network-only'
   })
+  const [retryPaidAction] = useMutation(RETRY_PAID_ACTION)
   const waitForWalletPayment = useWalletPayment()
   const invoiceHelper = useInvoice()
   const waitForQrPayment = useQrPayment()
   const client = useApolloClient()
   // innerResult is used to store/control the result of the mutation when innerMutate runs
   const [innerResult, setInnerResult] = useState(result)
+  const { wallets: walletDefs } = useWallets()
 
-  const waitForPayment = useCallback(async (invoice, { alwaysShowQROnFailure = false, persistOnNavigate = false, waitFor }) => {
-    let walletError
-    const start = Date.now()
+  // walletDefs shouldn't change on rerender, so it should be safe
+  const usableWallets = walletDefs
+    .map(w => w.def.name)
+    .map(name => useWallet(name))
+    .filter(w => !w.def.isAvailable || w.def.isAvailable())
+    .filter(w => w.config?.enabled && canSend(w))
+
+  const waitForActionPayment = useCallback(async (invoice, { wallet, alwaysShowQROnFailure = false, persistOnNavigate = false, waitFor }, originalResponse, action) => {
+    const walletErrors = []
+    let response = originalResponse
+
+    while (true) {
+      // we try all the sending wallets one by one
+      for (const wallet of usableWallets) {
+        try {
+          await waitForWalletPayment(invoice, waitFor, wallet)
+          console.log('paid with wallet', wallet.def.name)
+          return { invoice, response }
+        } catch (err) {
+          walletErrors.push(err)
+          console.log('could not pay with wallet', wallet.def.name, 'will try another one ...')
+        }
+      }
+
+      try {
+        await invoiceHelper.cancel(invoice)
```

Comment:
What if somehow the wallet failed but is still  trying to pay? (maybe an improper timeout?). Safer to wait for the invoice to be cancelled, so we don't double invoice.